components/bio-formats-tools/src/loci/formats/tools/ImageInfo.java

```diff
@@ -1141,8 +1142,14 @@ public static void main(String[] args) throws Exception {
 
       while (scanner.hasNext()) {
         newArgs[idx] = scanner.nextLine();
-	System.out.println("====% " + newArgs[idx]);
-        if (!new ImageInfo().testRead(newArgs)) System.exit(1);
+        System.out.println("====% " + newArgs[idx]);
+        try {
+        	new ImageInfo().testRead(newArgs);
+        }
+        catch (UnknownFormatException e) {
+			LOGGER.warn("Unknown file format: " + newArgs[idx]);
```

Comment:
`ImageInfo.testRead` can throw three types of exception:
- `FormatException` (of which `UnknowFormatException` is a subclass of)
- `IOException`
- `ServiceException`

What is the rationale for handling `UnknownFormatException` differently from the other exceptions?

Reply:
Thanks a lot @sbesson for your code review and for pointing out the other possible exceptions. I intended to do the least changes to the current behaviour. I can add exception handling for the other cases, too, i.e. to handle general FormatException and IOException. Could you please tell me, in which case a ServiceException would occur?

Reply:
You can read more about `ServiceException` in https://bio-formats.readthedocs.io/en/latest/developers/service.html. Typically this would happen when a dependency is missing.